src/org/exist/collections/MutableCollection.java

```diff
@@ -1258,6 +1258,9 @@ private void storeXMLInternal(final Txn transaction, final DBBroker broker, fina
         if(info.isCreating()) {
             info.getTriggers().afterCreateDocument(broker, transaction, document);
         } else {
+            final StreamListener listener = broker.getIndexController().getStreamListener(document, StreamListener.ReindexMode.REPLACE_DOCUMENT);
```

Comment:
@shabanovd You are mistaken.
Each `DBBroker` has an `IndexController`. Each `IndexController` has one `IndexWorker` per-index; therefore each DBBroker has a single `IndexWorker` per Index.

So far in all the index implementations, my own included, the `IndexWorker` has a single `StreamListener`. So `startReplaceDocument` will be called on the same `StreamListener` as the other events for the same document being processed by the same broker.